nav2_behaviors/plugins/drive_on_heading.cpp

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2022 Joshua Wallace
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <ctime>
+#include <memory>
+#include <utility>
+
+#include "drive_on_heading.hpp"
+#include "nav2_util/node_utils.hpp"
+
+using namespace std::chrono_literals;
+
+namespace nav2_behaviors
+{
+
+DriveOnHeading::DriveOnHeading()
+: TimedBehavior<DriveOnHeadingAction>(),
+  feedback_(std::make_shared<DriveOnHeadingAction::Feedback>())
+{
```

Comment:
I'm going to assume this entire file is exactly the same, not going to look through